staticd/static_nb_config.c

```diff
@@ -0,0 +1,1150 @@
+		if (!nh) {
+			char buf[SRCDEST2STR_BUFFER];
+
+			flog_warn(
+				EC_LIB_NB_CB_CONFIG_APPLY,
+				"%s : nh [%d:%s:%s:%s] nexthop creation failed",
+				srcdest_rnode2str(rn, buf, sizeof(buf)),
+				nh_type, ifname,
+				yang_dnode_get_string(args->dnode, "./gateway"),
+				nh_vrf);
+			return NB_ERR;
+		}
+		nb_running_set_entry(args->dnode, nh);
+		break;
+	}
+
+	return NB_OK;
+}
+
+static bool static_nexthop_destroy(struct nb_cb_destroy_args *args,
+				   const struct lyd_node *rn_dnode,
+				   struct stable_info *info)
+{
+	struct route_node *rn;
+	struct static_path *pi;
+	const struct lyd_node *pi_dnode;
+	struct static_nexthop *nh;
+	int ret;
+
+	nh = nb_running_unset_entry(args->dnode);
+	pi_dnode = yang_dnode_get_parent(args->dnode, "path-list");
+	pi = nb_running_get_entry(pi_dnode, NULL, true);
+	rn = nb_running_get_entry(rn_dnode, NULL, true);
+
+	ret = static_delete_nexthop(rn, pi, info->safi, info->svrf, nh);
+	if (!ret) {
+		char buf[SRCDEST2STR_BUFFER];
+
+		flog_warn(EC_LIB_NB_CB_CONFIG_APPLY,
+			  "%s : nh [%d:%s:%s:%s] nexthop destroy failed",
+			  srcdest_rnode2str(rn, buf, sizeof(buf)),
+			  yang_dnode_get_enum(args->dnode, "./nh-type"),
+			  yang_dnode_get_string(args->dnode, "./interface"),
+			  yang_dnode_get_string(args->dnode, "./gateway"),
+			  yang_dnode_get_string(args->dnode, "./vrf"));
+		return NB_ERR;
+	}
+
+	return NB_OK;
+}
+
+static int nexthop_mpls_label_stack_entry_create(struct nb_cb_create_args *args)
+{
+	struct static_nexthop *nh;
+	uint8_t id;
+
+	switch (args->event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+		break;
+	case NB_EV_APPLY:
+		nh = nb_running_get_entry(args->dnode, NULL, true);
+		id = yang_get_list_pos(args->dnode->parent) - 1;
+		nh->snh_label.label[id] = 0;
+		nh->snh_label.num_labels++;
+		break;
+	}
+
+	return NB_OK;
+}
+
+static int
+nexthop_mpls_label_stack_entry_destroy(struct nb_cb_destroy_args *args)
+{
+	struct static_nexthop *nh;
+	uint8_t id;
+
+	switch (args->event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+		break;
+	case NB_EV_APPLY:
+		nh = nb_running_get_entry(args->dnode, NULL, true);
+		id = yang_get_list_pos(args->dnode->parent) - 1;
+		nh->snh_label.label[id] = 0;
+		nh->snh_label.num_labels--;
+		break;
+	}
+
+	return NB_OK;
+}
+
+static int static_nexthop_mpls_label_modify(struct nb_cb_modify_args *args)
+{
+	struct static_nexthop *nh;
+	uint8_t id;
+
+	nh = nb_running_get_entry(args->dnode, NULL, true);
+	/* Mapping to array = list-index -1 */
+	id = yang_get_list_pos(args->dnode->parent) - 1;
+	nh->snh_label.label[id] = yang_dnode_get_uint32(args->dnode, NULL);
+
+	return NB_OK;
+}
+
+static int static_nexthop_mpls_label_destroy(struct nb_cb_destroy_args *args)
+{
+	struct static_nexthop *nh;
+	uint8_t id;
+
+	nh = nb_running_get_entry(args->dnode, NULL, true);
+	/* Mapping to array = list-index -1 */
+	id = yang_get_list_pos(args->dnode->parent);
+	nh->snh_label.label[id] = 0;
```

Comment:
You're not missing anything. In theory nothing needs to be done on this callback as `nexthop_mpls_label_stack_entry_destroy()` already takes care of doing a `nh->snh_label.num_labels--;`.

This callback wouldn't need to exist had the "label" leaf of the "mpls-label-stack/entry" list been marked as mandatory. But it isn't, and we can't change it since it's part of an IETF module (only option would be to define our own grouping for MPLS label stacks).

Reply:
done